### PR TITLE
119_create_scrapping_worker

### PR DESCRIPTION
**Pull request by ArturLabenskyy**
So finished create scrapping worker. it runs after server start and than every 7 days. it check if top songs collection in DB is empty or more than week is passed. Before updating top songs, it clears the old collection.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -49,16 +52,21 @@ app.use(errorHandler);
 const PORT = process.env.PORT || 5000;
 const NODE_ENV = process.env.NODE_ENV;
 
-connectDB();
+await connectDB();
+await scrapeTop10Songs();
+setInterval(() => {
+    clearTop10Songs();
```
Comment (yishain11) [inline server/server.js:58]: dont clear before you get new results to replace the old ones. what if the  scrapeTop10Songs function fails?

File: server/controllers/topSongsController.js [hunk 2]
```diff
@@ -9,120 +9,162 @@ import { dummySongsArray } from "../utils/createDummyData.js";
...
+        // run createTopSongs function if there no data in DB or more than week passed
+        if (!topSongs || (topSongs && checkIfAWeekPassed(topSongs[0].date))) {
+            console.log("top songs :", topSongs);
+            topSongs = axios
+                .post("http://localhost:5000/api/v1/harmony/topSongs")
+                .then((res) => {
+                    return;
+                })
+                .catch((error) => {
+                    console.log(error);
```
Comment (yishain11) [inline server/controllers/topSongsController.js:152]: use logger

File: server/controllers/topSongsController.js [hunk 3]
```diff
@@ -9,120 +9,162 @@ import { dummySongsArray } from "../utils/createDummyData.js";
...
+                });
+        }
+    } catch (error) {
+        console.log("try / catch error");
```
Comment (yishain11) [inline server/controllers/topSongsController.js:156]: use logger, and add relevant error details

File: server/controllers/topSongsController.js [hunk 4]
```diff
@@ -9,120 +9,162 @@ import { dummySongsArray } from "../utils/createDummyData.js";
...
+    }
+};
+
+const clearTop10Songs = async () => {
+    TopSongs.deleteMany({})
+        .then(() => {
+            console.log("Collection cleaned successfully.");
```
Comment (yishain11) [inline server/controllers/topSongsController.js:163]: logger

File: server/controllers/topSongsController.js [hunk 5]
```diff
@@ -9,120 +9,162 @@ import { dummySongsArray } from "../utils/createDummyData.js";
...
+        })
+        .catch((error) => {
+            console.error("Error cleaning collection:", error);
```
Comment (yishain11) [inline server/controllers/topSongsController.js:166]: logger

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Comment by ArturLabenskyy** [comment]
Review changes updated

**Pull request closed without merging**